specification/attestation.adoc

```diff
@@ -1,309 +1,885 @@
-| issuer | “<UDS_ID> or <CDI_ID>” UDS_ID, CD_ID are hex encoded lower case
-| validity | The validity values are populated as follows: notBefore can be 
-any time known to be in the past, and notAfter is set to the standard 
-value used to indicate no well-known expiry date, “99991231235959Z” per 
-RFC 5280.  
-| subject | “<CDI_ID>” where CDI_ID is hex encoded lower case
-| subjectPublicKeyInfo | When using ECDSA, per RFC 5480 (id-ecPublicKey)
-| issuerUniqueID | Not used
-| subjectUniqueID | Not used
-| extensions | Standard extensions are included as well as a custom TCG 
-extension which holds information about the measurements used to derive 
-CDI values. Both are described 
-below.                                                  
+|TCB |Hardware Elements |Software Elements
+
+.4+.^|Platform
+    <| HW RoT for boot, measurement and storage
+ .4+<| All M-mode firmwares, except for the TSM-driver
+    <| All CPU hardware logic, including MMU and caches
+    <| All SoC subsystems, including memory confidentiality, integrity and replay-protection for volatile memory
+    <| IOMMU and translation agents
+
+.2+.^| TSM
+.2+.^| N/A
+    <| TEE Security Manager (TSM) and its U-mode components
+    <| TSM-Driver that hosts a TEEI (with TH-ABI and TG-ABI security routines).
+
+     | TVM
+     | Directly assigned, TEE-IO compliant devices
+     | All TVM measured pages
 |===
 
-*CDI Standard Extensions*
+==== TVM Measurements
 
-|===
-| *Extension* | *Critical* | *Description*
-| authorityKeyIdentifier | non-critical | Contains only keyIdentifier set 
-to UDS_ID or previous CDI_ID
-| subjectKeyIdentifier | non-critical | Set to CDI_ID
-| keyUsage | critical | Contains only keyCertSign. Other CDI certificates 
-may be generated for other purposes for the TVM.
-| basicConstraints | critical | The cA field is set to TRUE. The 
-pathLenConstraint field is normally not included, but may be included 
-and set to zero if it is known that no additional DICE layers exist. 
-For example, for TVMs, this field may be set to zero.  
-|===
+At a high level, a TVM measurement is separated into a set of static
+measurements and a dynamic one.
+
+===== Static Measurements
+
+TVM static measurements are generated from the AP-TEE workload TCB elements
+involved in the TVM construction. Any TCB element that directly or indirectly
+supports a TVM generation must be measured into the TVM static measurement
+registers. Once a TVM is finalized, i.e. after the
+`sbi_tee_host_finalize_tvm()` TH-ABI is called, the TVM static measurements
+must no longer be extended.
+
+Static measurements are stored in dedicated measurement registers and an AP-TEE
+implementation must provide at least 1 and at most 8 of them. The static
+measurement registers cover platform, TSM & TVM specific TCB element
+measurements. The platform/TSM measurement registers may be stored in the
+hardware RoT for Measurement with platform bound life cycle. The TVM
```

Comment:
what do you imply by platform-bound life cycle?

Reply:
Yes, that's what I meant. I tried to rephrase, let me know if it's clearer now.